mail_service/worker_notify_user.py

```diff
@@ -156,6 +160,25 @@ def notify_user_with_linter_results(job_info):
             job.delete()
             continue
 
+    if "build_failed" in job_info:
+        if job_info["build_failed"]:
```

Comment:
I don't see an `else` clause to think there's some other situation being tackled here!